subsys/net/ip/net_context.c

```diff
@@ -2345,12 +2350,21 @@ static void set_pkt_hoplimit(struct net_pkt *pkt, const struct msghdr *msg_hdr)
 				net_pkt_set_ipv6_hop_limit(pkt, *(uint8_t *)CMSG_DATA(cmsg));
 				break;
 			}
-		} else {
-			if (cmsg->cmsg_len == CMSG_LEN(sizeof(int)) &&
-			    cmsg->cmsg_level == IPPROTO_IP &&
-			    cmsg->cmsg_type == IP_TTL) {
-				net_pkt_set_ipv4_ttl(pkt, *(uint8_t *)CMSG_DATA(cmsg));
-				break;
+		} else if (net_pkt_family(pkt) == AF_INET) {
+			if (!addr6 || (addr6 && !net_ipv6_addr_is_v4_mapped(&addr6->sin6_addr))) {
```

Comment:
Please change `!addr6` to `addr6 == NULL` as we try to comply to some Misra rule that says the short variant should only used if the variable is boolean.
Also change `addr6 &&` to `addr != NULL &&`